### PR TITLE
Fixes #24221 - Use precompiled regexp for percent-placeholder matching.

### PR DESCRIPTION
**Pull request by tricoder42**
See https://code.djangoproject.com/ticket/24221 for more info.

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

File: django/views/generic/edit.py [hunk 1]
```diff
@@ -12,6 +12,8 @@
 from django.views.generic.detail import (SingleObjectMixin,
                         SingleObjectTemplateResponseMixin, BaseDetailView)
 
+percent_placeholder = re.compile(r'%\([^\)]+\)')
```
Comment (timgraham) [inline django/views/generic/edit.py]: module level constants should be all caps

**Comment by timgraham** [comment]
merged in ea0ea7859a224225950a4df7c23eb3a7d823ddcd.

For future reference, commit message should be in past tense: "Fixed #XXXX -- Used ..." -- thanks!

**Comment by tricoder42** [comment]
Ok, thank you for review!

**Pull request closed without merging**